File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/DelayHandle/DelayPriorityHandleMgrBase.cpp
#include "MyProject.h"
#include "DelayPriorityHandleMgrBase.h"

MY_BEGIN_NAMESPACE(MyNS)

DelayPriorityHandleMgrBase::DelayPriorityHandleMgrBase()
{
	this.mDeferredAddQueue = new PriorityList();
	this.mDeferredAddQueue.setIsSpeedUpFind(true);
	this.mDeferredDelQueue = new PriorityList();
	this.mDeferredDelQueue.setIsSpeedUpFind(true);
}

void DelayPriorityHandleMgrBase::init()
{
	Super::init();
}

void DelayPriorityHandleMgrBase::dispose()
{
	if(null != this.mDeferredAddQueue)
	{
		this.mDeferredAddQueue.dispose();
		this.mDeferredAddQueue = null;
	}
	if (null != this.mDeferredDelQueue)
	{
		this.mDeferredDelQueue.dispose();
		this.mDeferredDelQueue = null;
	}

	Super::dispose();
}
	
MY_END_NAMESPACE

File: Project/MyProject/MyProject/Content/MyAsset/Lua/MyLua/Libs/DelayHandle/DelayPriorityHandleMgrBase.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

MLoader("MyLua.Libs.DataStruct.NoOrPriorityList.PriorityList.PriorityList");
MLoader("MyLua.Libs.DelayHandle.DelayNoOrPriorityHandleMgrBase");

-- 当需要管理的对象可能在遍历中间添加的时候，需要这个管理器
local M = GlobalNS.Class(GlobalNS.DelayNoOrPriorityHandleMgrBase);
M.clsName = "DelayPriorityHandleMgrBase";
GlobalNS[M.clsName] = M;

function M:ctor()
	self.mDeferredAddQueue = GlobalNS.new(GlobalNS.PriorityList);
	self.mDeferredAddQueue:setIsSpeedUpFind(true);
	self.mDeferredDelQueue = GlobalNS.new(GlobalNS.PriorityList);
	self.mDeferredDelQueue:setIsSpeedUpFind(true);
end

function M:init()
	M.super.init(self);
end

function M:dispose()
	M.super.dispose(M);
end

return M;